### PR TITLE
Add support for `cml comment <verb> --publish-url`

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
Required for https://github.com/iterative/cml.dev/issues/316
```
cml comment <create|update> --publish --publish-url=https://{identifier}.lambda-url.{region}.on.aws
```

### CODE REVIEW

File: src/cml.js [hunk 1]
```diff
@@ -208,7 +209,11 @@ class CML {
           );
           if (!triggerFile && watch) watcher.add(absolutePath);
           try {
-            node.url = await this.publish({ ...opts, path: absolutePath });
+            node.url = await this.publish({
```
Comment (tasdomas) [inline src/cml.js:212]: Should we have  a test for this?

**Review by tasdomas — APPROVED** [review]
LGTM, but we may want to have a test for this option.

File: src/cml.js (hunk 1 above)
Comment (0x2b3bfa0) [reply src/cml.js:212]: Would https://github.com/iterative/cml/pull/1166/commits/75bb629362c46765cd4d37e1a4cef817460229e2 be enough?

Comment (tasdomas) [reply src/cml.js:212]: won't we be uploading artifacts for every test run?

Comment (0x2b3bfa0) [reply src/cml.js:212]: *Cough,* yes, in negligible proportion when compared to the legacy `cml publish` tests. 😅 

https://github.com/iterative/cml/blob/dc6749c3fb79a5052cf3ebc3d77a9f4b940da7cd/bin/cml/asset/publish.test.js#L33-L109

**Review by tasdomas — APPROVED** [review]

**Comment by dacbd** [comment]
@0x2b3bfa0 I haven't tried this yet, know if you can save me an iteration; does the `cml comment` version of this flag respect the `CML_PUBLISH_URL` env that using `cml publish` does?

**Comment by 0x2b3bfa0** [comment]
@dacbd, no, it doesn't

**Comment by 0x2b3bfa0** [comment]
> _should it?_

No, we don't need any kind of backwards compatibility here

**Comment by 0x2b3bfa0** [comment]
> _`CML_COMMENT_PUBLISH_URL` is what yargs would infer?_

Yes

**Pull request merged**